beacon-chain/operations/attestations/prune_expired.go

```diff
@@ -84,3 +105,16 @@ func (s *Service) expiredPreDeneb(slot primitives.Slot) bool {
 	currentTime := uint64(prysmTime.Now().Unix())
 	return currentTime >= expirationTime
 }
+
+// Attestations for a slot before the returned slot are considered expired.
+func (s *Service) expirySlot() (primitives.Slot, error) {
+	currSlot := slots.CurrentSlot(s.genesisTime)
+	currEpoch := slots.ToEpoch(currSlot)
+	if currEpoch == 0 {
+		return 0, nil
+	}
+	if currEpoch < params.BeaconConfig().DenebForkEpoch {
+		currSlot.SubSlot(params.BeaconConfig().SlotsPerEpoch).Add(1)
+	}
+	return slots.EpochStart(currEpoch - 1)
```

Comment:
There's a bug here, `currSlot` is updated and then never used.  You may want to return early :joy: , but also there's no need to do all the `SubSlot`, and `Add` with a bunch of extra checks and jumps: you have already asserted that `DenebForkEpoch > currEpoch > 0 `  therefore you know you can simply subtract 31

Reply:
great find